services/users-api/lambdas/fetchReferenceCode.js

```diff
@@ -0,0 +1,82 @@
+import to from 'await-to-js';
+import { throwError } from '@helsingborg-stad/npm-api-error-handling';
+
+import config from '../../../config';
+import * as response from '../../../libs/response';
+import * as dynamoDb from '../../../libs/dynamoDb';
+import log from '../../../libs/logs';
+
+/**
+ * Get the referenceCode associated with the user that has the given email or PN
+ */
+export const main = async (event, context) => {
+  const query = event.pathParameters?.query;
+
+  if (!query) {
+    return response.failure({
+      status: 400,
+      message: 'Missing required parameter: "query"',
+    });
+  }
+
+  const isEmail = query.includes('@');
+
+  const action = isEmail ? 'query' : 'get';
+  let params;
+
+  if (isEmail) {
+    params = {
+      TableName: config.users.tableName,
+      IndexName: 'email-index',
+      KeyConditionExpression: 'email = :e_mail',
+      ExpressionAttributeValues: {
+        ':e_mail': `${query}`,
+      },
+    };
+  } else {
+    params = {
+      TableName: config.users.tableName,
+      Key: {
+        personalNumber: query,
+      },
+    };
+  }
+
+  const [error, userGetResponse] = await to(retrieveUser(action, params));
+  if (error) {
+    log.error(
+      'Fetch reference code request error',
+      context.awsRequestId,
+      'service-users-api-fetchReferenceCode-500',
```

Comment:
I wonder if the number at the end of this string `service-users-api-fetchReferenceCode-500` should be 001 instead of 500. Because it is the first occurrence of logging in the file.